src/test/test.cpp

```diff
@@ -481,12 +511,6 @@ void
 for_versions(std::vector<uint32> const& versions, Application& app,
              std::function<void(void)> const& f)
 {
-    uint32_t previousVersion = 0;
-    {
-        LedgerTxn ltx(app.getLedgerTxnRoot());
-        previousVersion = ltx.loadHeader().current().ledgerVersion;
-    }
-
     for (auto v : versions)
     {
         if (!gTestAllVersions &&
```

Comment:
Are you saying we don't need to check `gTestAllVersions` anymore? That's true, but it's more efficient (not that it matters here...) to check it since it means avoiding a linear lookup. I'm fine with removing the check though.

Reply:
Sorry, this comment was an error. Please ignore.

Reply:
Let me try to rewrite this comment in a way that makes sense.

Right now you're doing the following quadratic time algorithm. For every `v` in `versions`:

1. Check if we're testing all versions or if `v` is in `gVersionsToTest` (eg. `v` is a version we're testing)
2. Check if `gTestingVersion == v` (eg. `v` is currently being tested)
3. If (1) and (2) are true, perform the test

Now let's make the following observations. `gTestingVersion` is always in `gVersionsToTest` by construction. So if `gTestingVersion == v` then `v` is in `gVersionsToTest`. Given this and the fact that `gVersionsToTest` contains all versions when testing all versions, check (2) implies check (1). Therefore we should remove check (1). In this approach, the function becomes: if `gTestingVersion` is in `versions` then perform the test. This is very simple and has linear complexity.

Reply:
Ahhh yeah this makes sense. Fixed.